src/components/PageRouter.js

```diff
@@ -70,29 +70,41 @@ const findExactMatch = (branches) => {
 class PageRouter extends React.Component {
   constructor (props) {
     super(props)
-    this.handleKeyPress = this.handleKeyPress.bind(this)
+    this.handleKeyUp = this.handleKeyUp.bind(this)
+    this.handleKeyDown = this.handleKeyDown.bind(this)
     this.keyPressed = null
     this.previousPath = '/'
   }
 
-  handleKeyPress (event) {
+  handleKeyUp (event) {
     if (event.key === 'Escape' && this.keyPressed !== event.key) {
       this.keyPressed = event.key
       if (this.currentBranch.length > 1) {
-        this.props.redirectRoute(this.currentBranch[this.currentBranch.length - 2].match.url)
+        this.props.history.push(this.currentBranch[this.currentBranch.length - 2].match.url)
```

Comment:
What is benefit of `adding` instead of removing via `history.goBack()`

https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/history.md

Reply:
I do not know what benefit is better - creating new history or going back. But anyway we are using `Link` from `react-router` for closing the dialog (using `close` button), which adds new history entry. I think these two cases should have the same behaviour.

Reply:
Good point, it should be probably consistent.
Anyway, `removing` follows better my understanding of "going back". My understanding is sticked to the "Back button" in browser.

Does it make sense to refactor the `close` button this way?

Reply:
I would prefer to create new history. I think users are usually not expecting the page to go back in history stack while clicking the link. Also IMO I think it is a bit weird for dialog to close and then have the possibility to go forward again to the dialog. 

I would rather allow using `ESC` only for dialogs and not for all the pages.